lib/__tests__/init.test.ts

```diff
@@ -135,4 +135,42 @@ describe("init", () => {
     setUserToken.mockRestore();
     supportsCookies.mockRestore();
   });
+
+  describe("callback for userToken", () => {
+    it("should trigger callback when userToken is set with cookie support", () => {
+      const supportsCookies = jest
+        .spyOn(utils, "supportsCookies")
+        .mockReturnValue(true);
+
+      analyticsInstance.init({ apiKey: "***", appId: "XXX", region: "de" });
+      // Because cookie is enabled, anonymous token must be generated already.
+      expect(analyticsInstance._userToken).toBeTruthy();
+      expect(analyticsInstance._userToken.length).toBeGreaterThan(0);
+      const callback = jest.fn();
+      analyticsInstance.onUserTokenChange(callback, { immediate: true });
+      expect(callback).toHaveBeenCalledWith(analyticsInstance._userToken);
+      expect(callback).toHaveBeenCalledTimes(1);
+
+      analyticsInstance.setUserToken("abc");
```

Comment:
can you add a test for setUserToken to be called first?

Reply:
https://github.com/algolia/search-insights.js/pull/209/commits/f8d2c56c58fafa7eb6cd14885404a6b34917e7d5